BJS2-40683-Приглашение присоединиться к этапу.

nue-01 opened the pull request:
Выполнил основные требования, добавил в конфиг префикс ко всем url: api/v1.

nue-01 committed:
BJS2-40683-Приглашение присоединиться к этапу.
Выполнил основные требования, добавил в конфиг префикс ко всем url: api/v1.

Shakhaev commented on src/main/java/faang/school/projectservice/filters/impl/InvitationDescriptionFilterImpl.java:
```diff
@@ -0,0 +1,25 @@
+                                         StageInvitationFilterDto filterDto) {
+        return invitationStream.filter(stageInvitation ->
+                stageInvitation.getStatus().equals(StageInvitationStatus.REJECTED) &&
+                        stageInvitation.getDescription().contains(filterDto.getDescription()));
```
Привет! Метод contains чувтсвителен к регистру. Если будут разные регистры у строки description в StageInvitation и в паттерне фильтра description, то сравнение может не удасться даже в случае очевидного совпадения. Я у себя в коде использовал .toLowerCase() на обе строки, чтобы решить этот нюанс.

nue-01 replied:
Не думал об этом 
большое спасибо за фидбек
использовал toLowerCase() на оба описания чтобы получить идентичный вид подстрок

MiraGlaceon commented on src/main/resources/application.yaml:
```diff
@@ -25,6 +25,10 @@ spring:
         calculations_channel:
           name: calculations_channel
 
+  mvc:
```
Ты об этом сообщи ребятам обязательно, пусть тоже пользуются когда смержишь

nue-01 committed:
BJS2-40683-Приглашение присоединиться к этапу.
Исправления: в фильтре на проверку описания добавил приведение к одному регистру.

nue-01 committed:
Merge branch 'Medusa-master-stream7' into feature/BJS2-40683

nue-01 committed:
BJS2-40683-Приглашение присоединиться к этапу.
Исправления: подправил основную и тестовую структуру проекта для StageInvitation.

nue-01 committed:
Merge branch 'Medusa-master-stream7' into feature/BJS2-40683

# Conflicts:
#	src/main/java/faang/school/projectservice/dto/invitation/StageInvitationDto.java

nue-01 committed:
BJS2-40683-Приглашение присоединиться к этапу.
Исправления: решил конфликты, убрал дубликаты общих файлов.

nue-01 committed:
Merge branch 'Medusa-master-stream7' into feature/BJS2-40683

# Conflicts:
#	src/main/java/faang/school/projectservice/service/stage/StageService.java
#	src/test/java/faang/school/projectservice/validator/stage/StageServiceTest.java

nue-01 committed:
BJS2-40683-Приглашение присоединиться к этапу.
Исправления: решил конфликты.
